RInAppMessaging/Classes/Router.swift

```diff
@@ -24,6 +27,17 @@ internal class Router: RouterType {
         self.dependencyManager = dependencyManager
     }
 
+    func discardDisplayedCampaign() {
+        DispatchQueue.main.async {
```

Comment:
Is it because discardDisplayedCampaign is not executed on the main thread? or is it to fix UI issues?

Reply:
it's not executed on main thread. I think I event got a runtime warning in the xcode